src/compiler/checker.ts

```diff
@@ -2913,6 +2915,21 @@ namespace ts {
             }
         }
 
+        function baseTypeHasUnappliedOuterTypeParameters(type: Type): boolean {
+            const originalBaseType = <InterfaceType>type;
+            const originalTypeReference = <TypeReference>type;

```

Comment:
You only use these for the type parameters and type arguments. Might as well just declare variables for those instead.